os/board/rtl8721csm/src/component/common/file_system/ftl/ftl.c

```diff
@@ -1013,9 +1067,47 @@ uint32_t ftl_read(uint16_t logical_addr, uint32_t *value)
     return  ret;
 }
 
+#if defined(CONFIG_AMEBAD_TRUSTZONE) && (FTL_SECURE == 1)
+/* FTL Secure Save and Load API */
+uint32_t ftl_secure_save_to_storage(void *pdata_tmp, uint16_t offset, uint16_t size)
+{
+	uint32_t ret = 0;
+	uint8_t* tmp_buff = NULL;
+
+	tmp_buff = rtw_zmalloc(size);
+	if (tmp_buff == NULL) {
+		return 0x10;
+	}
+
+	up_allocate_secure_context(FTL_SECURE_CONTEXT_SIZE);
+	ret = ameba_ftl_save_to_storage(tmp_buff, pdata_tmp, offset, size);
+	if (ret != FTL_WRITE_SUCCESS) {
+		FTL_PRINTF(FTL_LEVEL_ERROR, "[ftl] ameba_ftl_save_to_storage Fail! ret: &x \n", ret);
+	}
+	up_free_secure_context();
+
+	rtw_mfree(tmp_buff, size);
+	return ret;
+}
+
+uint32_t ftl_secure_load_from_storage(void *pdata_tmp, uint16_t offset, uint16_t size)
+{
+	uint32_t ret = 0;
+
+	up_allocate_secure_context(FTL_SECURE_CONTEXT_SIZE);
+	ret = ameba_ftl_load_from_storage(pdata_tmp, offset, size);
+	if (ret != FTL_READ_SUCCESS) {
+		FTL_PRINTF(FTL_LEVEL_ERROR, "[ftl] ameba_ftl_load_from_storage Fail! ret: &x \n", ret);
+	}
+	up_free_secure_context();
+
+	return ret;
+}
+#endif
+
 // return 0 success
 // return !0 fail
-uint32_t ftl_save_to_storage(void *pdata_tmp, uint16_t offset, uint16_t size)
+uint32_t ftl_save_to_storage_i(void *pdata_tmp, uint16_t offset, uint16_t size)

```

Comment:
What does `i` means? I think this function name is bad to understand.
Please rename it to name related to the actual operation for better understanding.

Reply:
It is for non-secure API, save or load without encryption.
I will rename it to ftl_non_secure_save_to_storage()